docs/examples/use_cases/paddle/resnet50/dali.py

```diff
@@ -0,0 +1,257 @@
+# Copyright (c) 2022 NVIDIA Corporation.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import ctypes
+import os
+from dataclasses import dataclass
+import paddle
+import numpy as np
+from nvidia.dali.backend import TensorListCPU
+import nvidia.dali.fn as fn
+import nvidia.dali.types as types
+from nvidia.dali.pipeline import Pipeline
+from nvidia.dali.plugin.paddle import DALIGenericIterator
+from utils.mode import Mode
+from utils.utility import get_num_trainers, get_trainer_id
+
+
+@dataclass
+class PipeOpMeta:
+    crop: int
+    resize_shorter: int
+    min_area: float
+    max_area: float
+    lower: float
+    upper: float
+    interp: types.DALIInterpType
+    mean: float
+    std: float
+    output_dtype: types.DALIDataType
+    output_layout: str
+    pad_output: bool
+
+def create_dali_pipeline(batch_size, num_threads, device_id, data_dir, ops_meta,
+                         shard_id, num_shards, dali_cpu=False, is_training=True,
+                         seed=12):
+    pipeline = Pipeline(batch_size, num_threads, device_id, seed=seed)
+    with pipeline:
```

Comment:
It's a bit of a nitpick, but I wonder if we could just go here for pipeline_def decorator? It feels a bit like a vestige of class-based API (that would, for example, prevent us from running auto_augments).

```suggestion
@pipeline_def(seed=12)
def dali_pipeline(data_dir, ops_meta, shard_id, num_shards, dali_cpu=False, is_training=True):
```

Reply:
Done